src/compat/db_vdb.c

```diff
@@ -658,37 +671,48 @@ db_compile_statement_local (DB_SESSION * session)
 	}
     }
 
-  /* translate views or virtual classes into base classes */
-  statement_result = mq_translate (parser, statement);
-  if (!statement_result || pt_has_error (parser))
+  /*
+     the remote-DML of dblink must not execute mq_translate
+     because the query of remote-DML should be executed at remote server side
+   */
+  if (PT_IS_DBLINK_DML_QUERY (statement))
     {
-      pt_report_to_ersys_with_statement (parser, PT_SEMANTIC, statement);
-      return er_errid ();
+      ;
     }
-  statement = statement_result;
-
-  /* prefetch and lock translated real classes to avoid deadlock */
-  (void) pt_class_pre_fetch (parser, statement);
-  if (pt_has_error (parser))
+  else
     {
-      pt_report_to_ersys_with_statement (parser, PT_SYNTAX, statement);
-      return er_errid ();
-    }
+      /* translate views or virtual classes into base classes */
+      statement_result = mq_translate (parser, statement);
+      if (!statement_result || pt_has_error (parser))
+	{
+	  pt_report_to_ersys_with_statement (parser, PT_SEMANTIC, statement);
+	  return er_errid ();
+	}
+      statement = statement_result;
 
-  /* validate include_oid setting in the session */
-  if (session->include_oid)
-    {
-      if (mq_updatable (parser, statement) == PT_UPDATABLE)
+      /* prefetch and lock translated real classes to avoid deadlock */
+      (void) pt_class_pre_fetch (parser, statement);
+      if (pt_has_error (parser))
 	{
-	  if (session->include_oid == DB_ROW_OIDS)
-	    {
-	      (void) pt_add_row_oid (parser, statement);
-	    }
+	  pt_report_to_ersys_with_statement (parser, PT_SYNTAX, statement);
+	  return er_errid ();
 	}
-      else
+
+      /* validate include_oid setting in the session */
+      if (session->include_oid)
```

Comment:
session->include_oid is not a boolean.
```suggestion
      if (session->include_oid != DB_NO_OIDS)
```

Reply:
This is not related to this issue, but it looks need to fix for readability.